eval/src/vespa/eval/tensor/dense/dense_cell_range_function.h

```diff
@@ -0,0 +1,31 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/eval/eval/tensor_function.h>
+
+namespace vespalib::tensor {
+
+/**
+ * Tensor function creating a view to a continuous range of cells in
+ * another tensor. The value type will (typically) change, but the
+ * cell type must remain the same.
+ **/
+class DenseCellRangeFunction : public eval::tensor_function::Op1
+{
+private:
+    size_t _offset;
+    size_t _length;
+
+public:
+    DenseCellRangeFunction(const eval::ValueType &result_type,
+                           const eval::TensorFunction &child,
+                           size_t offset, size_t length);
+    ~DenseCellRangeFunction();
```

Comment:
I normally add override on destructors too. clion(clang) likes that.

Reply:
I will try to remember that. Added override to new classes.